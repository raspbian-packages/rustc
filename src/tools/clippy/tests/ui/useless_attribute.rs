#![feature(plugin)]
#![plugin(clippy)]
#![warn(useless_attribute)]

#[allow(dead_code, unused_extern_crates)]
extern crate clippy_lints;

// don't lint on unused_import for `use` items
#[allow(unused_imports)]
use std::collections;

// don't lint on deprecated for `use` items
mod foo { #[deprecated] pub struct Bar; }
#[allow(deprecated)]
pub use foo::Bar;

fn main() {}
