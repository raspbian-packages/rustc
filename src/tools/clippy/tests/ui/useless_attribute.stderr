error: useless lint attribute
 --> $DIR/useless_attribute.rs:5:1
  |
5 | #[allow(dead_code, unused_extern_crates)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(dead_code, unused_extern_crates)]`
  |
  = note: `-D useless-attribute` implied by `-D warnings`

error: aborting due to previous error

